### PR TITLE
Revert "Unpin grpcio for Ray Client (#37368)"

### PR DESCRIPTION
**Pull request by rkooo567**
Screen Shot 2023-07-14 at 9 28 53 AM
This reverts commit 61401dad861bff84e4c0b82f561ff539ac54b4ff.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — APPROVED** [review]

**Comment by jjyao** [comment]
The error introduced by the reverted PR

```


(22:46:10) INFO: From Testing //python/ray/tests:test_ray_init:
--
  | ==================== Test output for //python/ray/tests:test_ray_init:
  | Traceback (most recent call last):
  | File "/private/var/tmp/_bazel_ec2-user/6ab9ca4bc8b689ed9acff6044745e900/execroot/com_github_ray_project_ray/bazel-out/darwin-opt/bin/python/ray/tests/test_ray_init.runfiles/com_github_ray_project_ray/python/ray/tests/test_ray_init.py", line 6, in <module>
  | import grpc
  | File "/usr/local/opt/miniconda/lib/python3.8/site-packages/grpc/__init__.py", line 22, in <module>
  | from grpc import _compression
  | File "/usr/local/opt/miniconda/lib/python3.8/site-packages/grpc/_compression.py", line 20, in <module>
  | from grpc._cython import cygrpc
  | ImportError: dlopen(/usr/local/opt/miniconda/lib/python3.8/site-packages/grpc/_cython/cygrpc.cpython-38-darwin.so, 2): Symbol not found: _kCFStreamPropertySocketNativeHandle
  | Referenced from: /usr/local/opt/miniconda/lib/python3.8/site-packages/grpc/_cython/cygrpc.cpython-38-darwin.so
  | Expected in: flat namespace
  | in /usr/local/opt/miniconda/lib/python3.8/site-packages/grpc/_cython/cygrpc.cpython-38-darwin.so


```

**Review by pcmoritz — APPROVED** [review]

**Comment by rkooo567** [comment]
Looks like I need code owner approval. cc @richardliaw @ericl @edoakes

**Review by richardliaw — APPROVED** [review]

**Pull request merged**
